app/views/access_permissions/partials/_default_public_user_role_form.html.erb

```diff
@@ -13,28 +14,35 @@
         <div class="user-assignment-controls">
           <div class="dropdown pull-right">
-            <button class="btn btn-light btn-role-select" type="button" id="defaultPublicUserRole" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
-              <%= f.object.default_public_user_role.name %>
-              <span class="caret"></span>
-            </button>
-            <%= f.hidden_field :default_public_user_role_id, value: f.object.default_public_user_role.id, class: "default-public-user-role-id" %>
-            <ul class="dropdown-menu dropdown-menu-right user-assignment-dropdown" aria-labelledby="defaultPublicUserRole">
-              <% user_roles_collection.each do |role| %>
-                <li>
-                  <a href="#" data-turbolinks="false" class="user-role-selector" data-role-id="<%= role[1] %>">
-                    <%= role[0] %>
-                  </a>
-                </li>
-              <% end %>
-              <% if assignable.top_level_assignable?  %>
-                <li role="separator" class="divider" data-hook="support-dropdown-separator"></li>
-                <li>
-                  <a href="#" data-turbolinks="false" class="user-role-selector" data-role-id="">
-                    <%= t('access_permissions.remove_access') %>
-                  </a>
-                </li>
-              <% end %>
-            </ul>
+            <% if editable %>
+              <button class="btn btn-light btn-role-select" type="button" id="defaultPublicUserRole" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+                <%= f.object.default_public_user_role.name %>
+                <span class="caret"></span>
+              </button>
+              <%= f.hidden_field :default_public_user_role_id, value: f.object.default_public_user_role.id, class: "default-public-user-role-id" %>
+              <ul class="dropdown-menu dropdown-menu-right user-assignment-dropdown" aria-labelledby="defaultPublicUserRole">
+                <% user_roles_collection.each do |role| %>
+                  <li>
+                    <a href="#" data-turbolinks="false" class="user-role-selector" data-role-id="<%= role[1] %>">
+                      <%= role[0] %>
+                    </a>
+                  </li>
+                <% end %>
+                <% if assignable.top_level_assignable?  %>
+                  <li role="separator" class="divider" data-hook="support-dropdown-separator"></li>
+                  <li>
+                    <a href="#" data-turbolinks="false" class="user-role-selector" data-role-id="">
+                      <%= t('access_permissions.remove_access') %>
+                    </a>
+                  </li>
+                <% end %>
+              </ul>
+            <% else %>
+              <button class="btn btn-light btn-role-select disabled" type="button">
+                <%= f.object.default_public_user_role.name %>
+                <span class="caret"></span>
```

Comment:
do we need caret, if it is not editable?

Reply:
yes, it's the most reliable way to ensure proper text alignment, we hide it with CSS

Reply:
ahh